devicetypes/smartthings/rgbw-light.src/rgbw-light.groovy

```diff
@@ -186,16 +214,22 @@ def zwaveEvent(physicalgraph.zwave.Command cmd) {
 	[linkText: linkText, descriptionText: "$linkText: $cmd", displayed: false]
 }
 
-def buildOffOnEvent(cmd){
-	[zwave.basicV1.basicSet(value: cmd), zwave.switchMultilevelV3.switchMultilevelGet()]
-}
```

Comment:
I feel like it would be great to reduce the redundant code below and continue to use this function, but change the `basicSet` to the `switchMultilevelSet`, and bring the use of `commands` into this function. And maybe call it `buildSwitchCommand` or something a little more relevant. Then you only have to call `buildSwitchCommand(0xFF)` or `buildSwitchCommand(0x00)`. And hell, I guess `ON` and `OFF` could just be made in to `getSWITCH_ON_VALUE() { 0xFF }` and `getSWITCH_OFF_VALUE() { 0x00 }` or something like that.

It could even be possible to re-use that function elsewhere that `switchMultilevelSet` is used?

Reply:
Resolved in the updarte.